### ISSUE TITLE
Accessibility review of the cloud.gov Pages application

### ISSUE THREAD

**Issue by svenaas** [cloud-gov/product#2496]
This task was created as part of the planning task #2471 

Steps:
- [x] Do WAVE scan of each user page in https://pages.cloud.gov/
- [x] Assess and remediate any findings of level Error or Contrast Error
- [x] Basic exploration of keyboard navigation of https://pages.cloud.gov/
- [x] Basic exploration of screen reader experience of https://pages.cloud.gov/

**Comment by svenaas** [comment]
WAVE scan complete, now that cloud-gov/pages-core#4130 and cloud-gov/pages-core#4129 made it possible to reach all of the routes: 

- `/sites`
- `/sites/id/builds`
- `/sites/id/builds/build_id/logs`
- `/sites/id/published`
- `/sites/id/published/branch_name`
- `/sites/id/settings`
- `/organizations`
- `/organizations/id`
- `/settings`
- `/sites/new`

This scanning found several general accessibility errors and a large number (hundreds+) of theme-based contrast errors all of which represent WCAG AA and/or AAA failings. They are documented in cloud-gov/pages-core#4132 and cloud-gov/pages-core#4133 and I'm working on them.

**Comment by svenaas** [comment]
Keyboard navigation is looking good. I'm noticing two quirks, that I'm not too concerned about: 

1. There's no skip link to get past the main navigation. Given that there are only a handful of nav links and they are all important to the application interface, I think this is OK. 

2. I'm finding (on Chrome/Mac) that — unlike on https://cloud.gov/pages — focus doesn't reset to the top of the page when navigating between pages. Bearing in mind that I am not the intended user, I am finding this convenient in terms of generally putting the focus where I'm likely to want it. I'm not sure as of this writing whether this behavior is being provided as a helpful default by my browser or whether it has something to do with the page implementation. I'm also not sure whether it would be good to change this. 

I'm giving both of these points some consideration.

**Comment by svenaas** [comment]
After reviewing WCAG Success Criterion 2.4.1 and some guidance [from W3C](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-skip.html) and [from WebAIM](https://webaim.org/techniques/skipnav/) I think we can leave well enough alone for the time being. Because we use `header` and `footer` regions and use lists for our nav and an `h1` above the primary page content there are mechanisms for moving around the page and finding the most important elements. If we find this is presenting a barrier in the future we can certainly change it. 

It _would_ somewhat improve the semantics of the page if we started using a `main` region and an `aside` for the sidebar nav instead of the `div`s we have, but this would require fussier work with the CSS, and is not essential for accessibility (or SEO, since these pages are for logged-in users).

**Comment by svenaas** [comment]
I've found one shortcoming for screen reader users which it would be impactful to remediate: the Home link (to `/`) does not provide a "cloud.gov Pages" context (nor is that context provided elsewhere). I've documented this issue and a simple proposed fix at cloud-gov/pages-core#4136.

**Comment by svenaas** [comment]
And the merging of cloud-gov/pages-core#4137 closes this task out.
